compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -1840,7 +1842,22 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             let root_place =
                 self.prefixes(borrow.borrowed_place.as_ref(), PrefixSet::All).last().unwrap();
             let local = root_place.local;
-            match self.body.local_kind(local) {
+            let local_kind = self.body.local_kind(local);
+            if let LocalKind::Temp = local_kind {
+                // Mutable references to constants and functions will implicitly create
+                // temporaries, even though immutable references don't. We use these below to add a
+                // note about this.
+                match &self.body.local_decls[local].local_info {
+                    Some(box LocalInfo::FnDefRef) => {
+                        implicit_temporary = Some(("function", "function pointer"));
```

Comment:
it actually doesn't put the function pointer into a temporary, but creates a temporary of the function item type (a zst). Maybe it could literally mention the zero sized nature, as that should make it obvious that you get no data to mutate (in case someone expected a function pointer).